packages/financial-templates-lib/src/price-feed/DominationFinancePriceFeed.js

```diff
@@ -0,0 +1,252 @@
+const { PriceFeedInterface } = require("./PriceFeedInterface");
+const { parseFixed } = require("@ethersproject/bignumber");
+
+// An implementation of PriceFeedInterface that uses DominationFinance's API to retrieve prices.
+class DominationFinancePriceFeed extends PriceFeedInterface {
+  /**
+   * @notice Constructs the DominationFinancePriceFeed.
+   * @param {Object} logger Winston module used to send logs.
+   * @param {Object} web3 Instance used for Web3 utilities and conversions.
+   * @param {String} pair Representation of the pair the price feed is tracking.
+   *    The string should be the representation used by the DomFi API to identify this pair.
+   * @param {Integer} lookback How far in the past the historical prices will be available using getHistoricalPrice.
+   * @param {Object} networker Used to send the API requests.
+   * @param {Function} getTime Returns the Unix timestamp in seconds.
+   * @param {Integer} minTimeBetweenUpdates Min number of seconds between updates. If update() is called again before
+   *      this number of seconds has passed, it will be a no-op.
+   * @param {Number} priceFeedDecimals Number of decimals to use to convert price to wei.
+   * @param {Bool} invertPrice Indicates if prices should be inverted before returned.
+   * @param {Number} tickPeriod Number of seconds interval between price entries.
+   */
+  constructor(
+    logger,
+    web3,
+    pair,
+    lookback,
+    networker,
+    getTime,
+    minTimeBetweenUpdates,
+    invertPrice,
+    priceFeedDecimals = 18,
+    tickPeriod = 60
+  ) {
+    super();
+    this.logger = logger;
+    this.web3 = web3;
+
+    this.pair = pair;
+    this.lookback = lookback;
+    this.tickPeriod = tickPeriod;
+    this.networker = networker;
+    this.getTime = getTime;
+    this.minTimeBetweenUpdates = minTimeBetweenUpdates;
+    this.invertPrice = invertPrice;
+
+    this.toBN = this.web3.utils.toBN;
+    this.convertPriceFeedDecimals = number => {
+      // Converts price result to wei
+      // returns price conversion to correct decimals as a big number
+      return this.toBN(parseFixed(number.toString(), priceFeedDecimals).toString());
+    };
+  }
+
+  get _priceUrl() {
+    return `https://api.domination.finance/api/v0/price/${this.pair}`;
+  }
+
+  get _historicalPricesUrl() {
+    return `https://live.domination.finance/api/v0/history/${this.pair}?tick=${this.tickPeriod}s&range=${this.lookback}s`;
+  }
+
+  getCurrentPrice() {
+    return this.invertPrice ? this._invertPriceSafely(this.currentPrice) : this.currentPrice;
+  }
+
+  getHistoricalPrice(time, verbose = false) {
+    if (this.lastUpdateTime === undefined) {
+      return undefined;
```

Comment:
Usually we return null for situations when price cannot be computed. Undefined should only be returned if the function goes down an unexpected path

Reply:
This was the verbatim from the `CryptoWatchPriceFeed.js`. I've updated it there as well.